stats.go

```diff
@@ -139,7 +168,11 @@ func (t *timer) Start() Stopwatch {
 }
 
 func (t *timer) Record(interval time.Duration) {
-	t.reporter.ReportTimer(t.name, t.tags, interval)
+	if t.cachedTimer != nil {
+		t.cachedTimer.ReportTimer(interval)
+	} else {
+		t.reporter.ReportTimer(t.name, t.tags, interval)
+	}
```

Comment:
I tried this.

```
raynos at raynos-ThinkPad-T440p  
~/gocode/src/github.com/uber-go/tally on cached-reporter*
$ go test -bench BenchmarkTimerReport -run _NONE_ -cpuprofile cpuprofile.out -benchmem -benchtime 7s -cpu 2 .
BenchmarkTimerReport-2   	2000000000	         6.02 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/uber-go/tally	12.633s
raynos at raynos-ThinkPad-T440p  
~/gocode/src/github.com/uber-go/tally on cached-reporter*
$ go test -bench BenchmarkTimerReport -run _NONE_ -cpuprofile cpuprofile.out -benchmem -benchtime 7s -cpu 2 .
BenchmarkTimerReport-2   	1000000000	         9.24 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/uber-go/tally	10.196s
```

It looks like the inline if statement is faster then the bound function call. Adding an extra function call takes 9/10ns instead of 5/6ns.